test/unit-test/test_lib/test_vis.py

```diff
@@ -73,69 +76,80 @@ def test_generate_probability_distribution(self):
         self.assertEqual(x, -1)
         self.assertEqual(freq, -1)
 
-    @data(
-        (util.load_from_disk(current_dir + "/data/conv_len"), util.load_from_disk(current_dir + "/data/conv_len_fit")))
-    @unpack
-    def test_exponential_curve_fit_and_plot(self, data, expected_result):
-        output = vis.exponential_curve_fit_and_plot(data, current_dir, "exponential_plot_test")
-        # remove the plot created
-        os.remove(current_dir + '/exponential_plot_test.png')
+    @patch("lib.vis.generate_probability_distribution", autospec=True)
+    @patch("lib.util.exponential_curve_func", autospec=True)
+    @patch("lib.vis.curve_fit", autospec=True)
+    def test_exponential_curve_fit_and_plot(self, mock_curve_fit, mock_curve_func, mock_probability_distribution):
+        data = util.load_from_disk(self.test_data_dir + "/vis/conv_len")
+        expected_result = util.load_from_disk(self.test_data_dir + "/vis/conv_len_fit")
+        mock_curve_fit.return_value = util.load_from_disk(self.test_data_dir + "/vis/curve_fit")
+        mock_curve_func.side_effect = self.mock_exponential_curve_func
+        mock_probability_distribution.side_effect = self.mock_generate_probability_distribution
+
+        output = vis.exponential_curve_fit_and_plot(data, self.test_data_dir, "exponential_plot_test")
+        os.remove(self.test_data_dir + '/exponential_plot_test.png')
         assert np.allclose(output, expected_result)
```

Comment:
The mocks need to be verified. Also @kaushik-rohit is using underscore suffix to write expected variables, ex: result_. It would be good to stick to a convention.

Reply:
i would prefer sticking to `expected_$varaible` convention because that's something been used widely in the project. If the general convention(in the python community) is of using `result_`,(and if we agree using it) We'll have to make changes across the project.